### PR TITLE
Remove batch functionality from sync-changeset subsystem

### PR DESCRIPTION
**Pull request by mrnugget**
@eseliger and I discovered that we never load/sync more than one changeset at once. The batch functionality is a leftover from a year ago, when we synced multiple changesets in the same call.

So after having to deal with "batch functionality and not-found-errors" in https://github.com/sourcegraph/sourcegraph/pull/14771 I did a little spike to see what it would take to remove this code.

Here we go.

Reviewers: what do you think?

### CODE REVIEW

File: cmd/repo-updater/repos/github.go [hunk 1]
```diff
@@ -235,31 +235,26 @@ func (s GithubSource) UndraftChangeset(ctx context.Context, c *Changeset) error
...
-			Number:        number,
-		}
+	pr := &github.PullRequest{
+		RepoWithOwner: repo.NameWithOwner,
+		Number:        number,
 	}
 
-	err := s.client.LoadPullRequests(ctx, prs...)
-	if err != nil {
+	fmt.Printf("pr=%+v\n", pr)
```
Comment (eseliger) [inline cmd/repo-updater/repos/github.go]:
```suggestion
```

File: cmd/repo-updater/repos/gitlab.go [hunk 2]
```diff
@@ -361,34 +361,32 @@ func (s *GitLabSource) CloseChangeset(ctx context.Context, c *Changeset) error {
 	return nil
 }
 
-// LoadChangesets loads the given merge requests from GitLab and updates them.
+// LoadChangeset loads the given merge requests from GitLab and updates them.
 // Note that this is an O(n) operation due to limitations in the GitLab REST
 // API.
```
Comment (eseliger) [inline cmd/repo-updater/repos/gitlab.go:366]:
```suggestion
// LoadChangeset loads the given merge request from GitLab and updates it.
```

File: enterprise/internal/campaigns/syncer.go [hunk 3]
```diff
@@ -513,187 +513,98 @@ func (s *ChangesetSyncer) SyncChangeset(ctx context.Context, id int64) error {
...
-	if len(cs) == 0 {
-		return nil
+func SyncChangeset(ctx context.Context, repoStore RepoStore, syncStore SyncStore, sourcer repos.Sourcer, c *campaigns.Changeset) (err error) {
+	s, err := buildChangesetSource(ctx, repoStore, sourcer, c)
+	if err != nil {
+		return err
 	}
 
-	bySource, err := groupChangesetsBySource(ctx, repoStore, sourcer, cs...)
+	rs, err := repoStore.ListRepos(ctx, repos.StoreListReposArgs{
```
Comment (eseliger) [inline enterprise/internal/campaigns/syncer.go:527]: man how many times have I seen this one

File: enterprise/internal/campaigns/resolvers/changeset_counts_test.go [hunk 4]
```diff
@@ -201,8 +201,8 @@ func TestChangesetCountsOverTimeIntegration(t *testing.T) {
 	apitest.MustExec(actor.WithActor(context.Background(), actor.FromUser(userID)), t, s, input, &response, queryChangesetCountsConnection)
 
 	wantCounts := []apitest.ChangesetCounts{
-		{Date: marshalDateTime(t, daysBeforeEnd(5)), Total: 0, Open: 0},
-		{Date: marshalDateTime(t, daysBeforeEnd(4)), Total: 1, Open: 1, OpenPending: 1},
+		{Date: marshalDateTime(t, daysBeforeEnd(5)), Total: 0, Open: 0, OpenPending: 0},
+		{Date: marshalDateTime(t, daysBeforeEnd(4)), Total: 1, Open: 0, OpenPending: 0},
```
Comment (eseliger) [inline enterprise/internal/campaigns/resolvers/changeset_counts_test.go:205]: This change seems dubious to me, idea why that happened?

**Review by eseliger — APPROVED** [review]

**Comment by mrnugget** [comment]
@sourcegraph/campaigns Erik and I finished this PR together in a Zoom-pairing-session. We're going to merge this as soon as the build passes. No more reviews necessary. The code is much simpler and easier to understand without the batch functionality, so no need to keep it.

**Pull request merged**